examples/src/main/java/com/squareup/moshi/recipes/ReadJsonListKt.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi.recipes
+
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import com.squareup.moshi.recipes.models.Card
+
+internal class ReadJsonListKt {
+
+  private val jsonString = """
+    [{"rank": "4",
+       "suit": "CLUBS"
+     },
+     {"rank": "A",
+      "suit": "HEARTS"
+      },
+     {"rank": "J",
+      "suit": "SPADES"
+     }]
+  """.trimMargin()
+
+  fun readJsonList() {
+    val moshi = Moshi.Builder().build()
+    val listOfCardsType = Types.newParameterizedType(
+      List::class.java,
+      Card::class.java
+    )
```

Comment:
Please use the reified extension we have for adapter lookups

Reply:
Could you please suggest the change? I am not successful using the Kotlin `typeOf` API. I was referring to your comment https://github.com/square/moshi/issues/1072#issuecomment-574474340.

```
    val cardType = typeOf<List<Card>>()
    val jsonAdapter = moshi.adapter(cardType)
```

Reply:
Take a look at DualKotlinTest for an example of what I'm referring to

Reply:
Thanks! Done.